src/astro/databases/snowflake.py

```diff
@@ -57,6 +76,252 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
+        """
+        Creates a stage name based on source_file path and storage integration provided.
+
+        @param source_file: Source file
+        @param snowflake_hook: Snowflake hook
+        @return: Stage name created
+        """
+        # Here,
+        # To copy dataset from gcs_path = gs://sample-bucket/workspace/sample.csv" to snowflake table
+        # where stage is WORKSPACE_STAGE, following steps has to be completed.
+
+        # The path of stage should be to the folder the file is in. Command to create STAGE would look
+        # like following:
+        # CREATE OR REPLACE STAGE <warehouse_name>.WORKSPACE_STAGE URL='gcs://sample-bucket/workspace'
+        # FILE_FORMAT=(TYPE=CSV, TRIM_SPACE=TRUE) COPY_OPTIONS=(MATCH_BY_COLUMN_NAME=CASE_INSENSITIVE)
+        # storage_integration = <storage_integration_name>;
+
+        # COPY INTO query would be:
+        # COPY INTO sample_table from '@WORKSPACE_STAGE/sample.csv';
+        FILE_FORMAT_MAPPING = {
+            FileType.CSV: "CSV",
+            FileType.PARQUET: "PARQUET",
+            FileType.NDJSON: "JSON",
+        }
+        snowflake_query_file_storage_mapping = {"gs": "gcs", "s3": "s3"}
+        stage_name = (
+            "stage_"
+            + random.choice(string.ascii_lowercase)
+            + "".join(
+                random.choice(string.ascii_lowercase + string.digits) for _ in range(7)
+            )
+        )
+
+        parsed_url = urlparse(source_file.path, allow_fragments=False)
+        bucket_type = snowflake_query_file_storage_mapping.get(
+            parsed_url.scheme.lower(), "gcs"
+        )
+
+        location_url = (
+            bucket_type + "://" + parsed_url.netloc + os.path.dirname(parsed_url.path)
+        )
+        if get_filetype(source_file.path) == FileType.CSV:
+            # match_by_column_name option is not supported for file format CSV
+            create_stage_sql = (
+                "CREATE OR REPLACE STAGE {} URL='{}' FILE_FORMAT=(TYPE={}, TRIM_SPACE=TRUE) "
+                "storage_integration = {};".format(
+                    stage_name,
+                    location_url,
+                    FILE_FORMAT_MAPPING.get(get_filetype(source_file.path)),
+                    self.integration,
+                )
+            )
+        else:
+            # match_by_column_name option is required for PARQUET and NDJSON formats
+            create_stage_sql = (
+                "CREATE OR REPLACE STAGE {} URL='{}' FILE_FORMAT=(TYPE={}, TRIM_SPACE=TRUE) "
+                "COPY_OPTIONS=(MATCH_BY_COLUMN_NAME=CASE_INSENSITIVE) storage_integration = {};".format(
+                    stage_name,
+                    location_url,
+                    FILE_FORMAT_MAPPING.get(get_filetype(source_file.path)),
+                    self.integration,
+                )
+            )
+        execution_info = snowflake_hook.run(create_stage_sql, autocommit=True)
```

Comment:
It is probably worth it for us to capture any exception here related to the user not having permissions to create a stage in snowflake, explaining which set of permissions is necessary and explaining how they can disable this behaviour (creating stage) if they don't have permissions for this.

Reply:
Makes sense. Adding the exception handling.

Reply:
[Done]([Add exception handling and change return types](https://github.com/astronomer/astro-sdk/pull/487/commits/863c4e9c8160f91412565a52aff93939f680638d))